src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -211,6 +222,41 @@ public SAMFileHeader getHeaderForReads() {
         }
     }
 
+    public void writeReads(final JavaSparkContext ctx, final String outputFile, JavaRDD<GATKRead> reads) {
+        try {
+            ReadsSparkSink.writeReads(ctx, outputFile, reads, readsHeader, shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE, getRecommendedParallelism());
+        } catch (IOException e) {
+            throw new GATKException("unable to write bam: " + e);
+        }
+    }
+
+    /**
+     * Return the recommended parallelism for a pipeline processing the reads. The parallelism is
+     * calculated by finding the total size (in bytes) of all the files in the input path, then
+     * dividing by the parallelism split size (determined by {@link #getParallelismPartitionSize()}.
+     * @return the recommended parallelism, or <code>defaultParallelism</code> if the size of the input
+     * cannot be determined
+     */
```

Comment:
Mention in the docs whether tools should ever override this method, and if so under what circumstances.